capstone_project/src/main/java/com/google/sps/data/DeletePostService.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.sps.data;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Service for deleting posts that were created longer than 24 hours ago. Note: This uses google
+ * cloud datastore since a cloud function does not have access to app engine datstore properties.
+ * Cloud datastore uses the same database as app engine. The only differences are in API usages and
+ * methods.
+ */
+public class DeletePostService {
+  // Represents one day in milliseconds
+  private final long ONE_DAY = 86400000;
+  private Datastore datastore;
+  private Clock clock;
+
+  public DeletePostService(Datastore datastore, Clock clock) {
+    this.datastore = datastore;
+    this.clock = clock;
+  }
+
+  // Retrieves a query of all posts entities in ascending order based on timestamp. Returns a list
+  // of entities of all posts inside of datastore in sorted ascending order.
+  public List<Entity> retrieveAllPostsAscending() {
```

Comment:
Why is this method public? It doesn't seem like a DeletePostService should support a method that lists out posts.

It also isn't used anywhere.

Reply:
Done